php_phongo.c

```diff
@@ -1998,24 +2008,13 @@ PHP_GINIT_FUNCTION(mongodb)
 	ZEND_TSRMLS_CACHE_UPDATE();
 #endif
 #endif
-	mongodb_globals->debug_fd = NULL;
+	memset(mongodb_globals, 0, sizeof(zend_mongodb_globals));
 	mongodb_globals->bsonMemVTable = bsonMemVTable;
-
+	/* Initialize HashTable for persistent clients */
+	zend_hash_init_ex(&mongodb_globals->clients, 0, NULL, php_phongo_client_dtor, 1, 0);
```

Comment:
It's possible that the root cause was not GINIT vs. MINIT, but rather previously fetching `mongodb_globals->clients` through our `MONGODB_G()`, since that uses [`ZEND_MODULE_GLOBALS_ACCESSOR()`](https://github.com/php/php-src/blob/php-7.0.14/Zend/zend_API.h#L162) on ZTS. For non-ZTS builds, the macro directly accesses the struct field, as we're doing here.

That said, other extensions initialize their persistent hashes in GINIT (see: [pgsql](https://github.com/php/php-src/blob/php-7.0.14/ext/pgsql/pgsql.c#L1098)), so I'd consider it good practice.